### PR TITLE
[DOC] Update Network Policies

### LINKED ISSUE #5765: Update network policy documentation
[Network policy documentation](https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-network-policies.html#k8s-network-policies-operator-isolation) states the K8S control plane IP can be retrieved using `kubectl cluster-info | grep master` while it is not working anymore.

### PR DESCRIPTION
**Pull request by barkbay**
Fixes #5765

This PR also:
* Fixes the operator label name which is now `app.kubernetes.io/name`
* Adds a sample for Agent and Fleet Server

### CODE REVIEW

**Review by thbkrkr — DISMISSED** [review]
There are other references of the old `control-plane` label in the webhook documentation which we can fix here or in another PR.

```diff
diff --git a/docs/operating-eck/webhook.asciidoc b/docs/operating-eck/webhook.asciidoc
index 344317542..15ed8a004 100644
--- a/docs/operating-eck/webhook.asciidoc
+++ b/docs/operating-eck/webhook.asciidoc
@@ -219,7 +219,7 @@ metadata:
 spec:
   podSelector:
     matchLabels:
-      control-plane: elastic-operator
+      app.kubernetes.io/name: elastic-operator
   ingress:
   - from: []
     ports:
@@ -245,7 +245,7 @@ metadata:
 spec:
   podSelector:
     matchLabels:
-      control-plane: elastic-operator
+      app.kubernetes.io/name: elastic-operator
```

**Comment by barkbay** [comment]
> So maybe let's keep that out of this PR?

Good catch! I added a comment in a "Prerequisites" section. The reason is that it is unfortunately not correct in both cases 😕
Let me know what you think.

@alaudazzi I would be glad to have your thoughts too, thanks 🙇

File: docs/advanced-topics/network-policies.asciidoc [hunk 1]
```diff
@@ -32,7 +33,35 @@ kubectl label namespace team-a eck.k8s.elastic.co/tenant=team-a
 kubectl label namespace team-b eck.k8s.elastic.co/tenant=team-b
 ----
 
+== Prerequisites
 
+To set up the network policies correctly you must know the operator Pod selector and the Kubernetes API server IP. They may vary depending on your environment and how the operator has been installed.
+
+=== Operator Ood selector
```
Comment (thbkrkr) [inline docs/advanced-topics/network-policies.asciidoc:40]:
```suggestion
=== Operator Pod selector
```

File: docs/advanced-topics/network-policies.asciidoc [hunk 2]
```diff
@@ -32,7 +33,35 @@ kubectl label namespace team-a eck.k8s.elastic.co/tenant=team-a
...
+
+The operator Pod labels depends on how the operator has been installed, refer to the following table to know what label name should be used in the network policies.
```
Comment (thbkrkr) [inline docs/advanced-topics/network-policies.asciidoc:42]:
```suggestion
The operator Pod label depends on how the operator has been installed, refer to the following table to know what label name should be used in the network policies.
```

**Comment by thbkrkr** [comment]
> Good catch! I added a comment in a "Prerequisites" section. The reason is that it is unfortunately not correct in both cases confused. Let me know what you think.

Yes, that's already not correct, but I would have waited for #5788 to fix it properly, probably by also setting the label `app.kubernetes.io/name` when installing via the raw manifest so that we use only one label in the code and in the document.

But we can keep your doc update as an intermediate fix :+1:.

**Comment by barkbay** [comment]
> Yes, that's already not correct, but I would have waited for https://github.com/elastic/cloud-on-k8s/issues/5788 to fix it properly

My concern is that the documentation in its current state might be misleading for users who have already deployed the operator. I agree we need to fix it, the intent here is to update the documentation for affected and supported versions.

**Review by thbkrkr — APPROVED** [review]
LGTM

**Pull request merged**